### ISSUE TITLE
[Issue]: RuntimeError: The GPU device instance has been suspended. Use GetDeviceRemovedReason to determine the appropriate action.

### ISSUE THREAD

**Issue by Lumpsack** [vladmandic/automatic#1696]
### Issue Description

Hi,
Just recently, I am unable to generate any images, and receive the below error.
I am running AMD so semi-used to issues, but this one is new.
I have so far:

- Rebooted
- Reset driver after issue and trued again
- Used the re-install flag to try and make sure no config issues remaining from older versions.
- Ran using --safe

The problem presists

![image](https://github.com/vladmandic/automatic/assets/127429289/b52cab54-b55e-49c0-904e-ee6d121ddd3e)
![image](https://github.com/vladmandic/automatic/assets/127429289/0eb40445-e85c-4349-b2ea-c1b199d62de1)

### Version Platform Description

Platform: Windows 11 Home H222 : 22621.1992
Browser: Edge : Version 114.0.1823.82 (Official build) (64-bit)
Driver: AMD Adrenaline 23.7.1 (latest)
Card : AMD 7900XTX

10:29:38-932577 INFO     Starting SD.Next
10:29:38-954082 INFO     Python 3.10.7 on Windows
10:29:39-083826 INFO     Version: e4463389 Fri Jul 14 21:56:10 2023 -0400
10:29:39-803684 INFO     Using DirectML Backend
10:29:54-675856 INFO     Torch 2.0.0+cpu
10:29:54-893709 INFO     Torch backend: DirectML (torch-directml 0.2.0.dev230426)
10:29:55-164176 INFO     Torch detected GPU: AMD Radeon RX 7900 XTX
10:29:55-910388 INFO     Verifying requirements
10:29:55-928423 WARNING  Package wrong version: timm 0.4.12 required 0.6.13
10:29:55-929922 INFO     Installing package: timm==0.6.13
10:30:06-339301 INFO     Verifying packages
10:30:06-341300 INFO     Verifying repositories
10:30:09-739717 INFO     Verifying submodules
10:30:32-798649 INFO     Extension installed packages: clip-interrogator-ext ['clip-interrogator==0.6.0']
10:31:12-912054 INFO     Extension installed packages: sd-webui-roop ['onnxruntime==1.15.0']
10:31:14-350973 INFO     Extensions enabled: ['clip-interrogator-ext', 'LDSR', 'Lora', 'ScuNET',
                         'sd-extension-system-info', 'sd-webui-agent-scheduler', 'sd-webui-model-converter',
                         'stable-diffusion-webui-images-browser', 'stable-diffusion-webui-rembg', 'SwinIR',
                         'a1111-sd-webui-tagcomplete', 'adetailer', 'canvas-zoom', 'ImageReward', 'OneButtonPrompt',
                         'sd-dynamic-prompts', 'sd-face-editor', 'sd-model-preview-xd', 'sd-webui-aspect-ratio-helper',
                         'sd-webui-regional-prompter', 'sd-webui-roop', 'Stable-Diffusion-Webui-Civitai-Helper',
                         'ultimate-upscale-for-automatic1111']
10:31:14-352973 INFO     Verifying packages
10:31:14-354472 INFO     Updating Wiki
10:31:14-994691 INFO     Extension preload: 0.0s D:\Vlad\automatic\extensions-builtin
10:31:14-996689 INFO     Extension preload: 0.0s D:\Vlad\automatic\extensions
10:31:15-010690 INFO     Server arguments: ['--use-directml', '--experimental', '--upgrade']
No module 'xformers'. Proceeding without it.
10:31:20-284468 INFO     Pipeline: Backend.ORIGINAL
10:31:21-663545 INFO     Libraries loaded
10:31:21-665044 INFO     Using data path: D:\Vlad\automatic
10:31:21-672545 INFO     Available VAEs: D:\Vlad\automatic\models\VAE 4
10:31:22-083966 INFO     Available models: D:\Vlad\automatic\models\Stable-diffusion 54

### Acknowledgements

- [X] I have read the above and searched for existing issues

**Comment by vladmandic** [comment]
That's a new one for me as well. Hope that someone with AMD can chime in as I can't reproduce.

**Comment by Lumpsack** [comment]
Thanks for checking - for whoever picks this up I've also now tried renaming the venv folder and letting it rebuild - no joy so renamed back.
Also, I believe the issue is DirectX (DirectML?) related, judging by a web search.

**Comment by vladmandic** [comment]
that's how it looks to me too, its several layers below the app itself.

**Comment by UnDiff** [comment]
I can't reproduce the error, but I have encountered the error with other people several times. Since I've been around Unreal Engine for a while, I've seen this message from time to time.

Not always, but often the problem was with the graphics card itself. You should check if the graphics card was slightly overclocked with the AMD tool or another tool and adjust it down a bit if necessary. Check the temperature, are all fans still running, is the graphics card too warm? 

As I said, not always, but very often this problem was related to these causes in the last 4 years.

**Comment by JPNorton** [comment]
Thanks for the reply, I can check when I'm on the pc, but this happens even if I open nothing else after a restart then try to generate something.
It's not overclocked and I can go on to play a game afterwards without issue.
I'm not sure why this only affects SD

**Comment by Lumpsack** [comment]
Hi again,
I just fired it up again to test, heres a screenshot showing the setting and confirming no overclock along with the temp etc and the GPU usage from task manager.

![image](https://github.com/vladmandic/automatic/assets/127429289/a1718733-34fc-429e-9a8c-4625b0dfbe57)

**Comment by Lumpsack** [comment]
Follow up - despite it seeming slow by comparison I ranTiger's Auto1111 and it worked, heres a shot of the monitors as reference.
![image](https://github.com/vladmandic/automatic/assets/127429289/d4b9c0cc-1962-464e-b031-c2e09c7d33f8)

**Comment by UnDiff** [comment]
You're right, it must be something else. The driver version also works fine for me. 

Especially because I now know which graphics card you have. The graphics card is potent enough. the Temps also. I would update to the latest version of SD Next, omit --experimental and disable all plugins except Lora, lycoris and systeminfo and restart. 

Or run a new git clone in a new temporary extra folder, start the installer with --use-directml and then test only with lora and lycoris after a reboot. This has helped me several times.

With me runs with
Python 3.10.11 on Windows 11 Prof 32 GB
Version: da11f32d SD Next
Same driver
AMD Radeon RX 5700 XT 8 GB (a real cucumber for SD^^)
with --debug --use-directml --upgrade --medvram
all is well.

**Comment by UnDiff** [comment]
Really strange. The only thing I can say is that Doggettx is currently giving me problems again and I have switched back to sub-quadratic. 

But don't worry, I also have both installed to switch, there is always something :-) Thank god, that means that it is currently being worked on hard^^

**Comment by TeutonJon78** [comment]
I just hit this for the first time debugging a different issue. @Lumpsack and @UnDiff are you both using the experimental autocast? What optimizer are you using when you hit the problem?

The time it triggered for me was starting up SD.next before the system had finished booting completely, and I wonder if it's catching the GPU/driver/DirectX is some sort of intermediate state.

**Comment by Lumpsack** [comment]
Sorry if I'm being dumb, but I don't understand the question, I'm not sure what an optimiser is in SD context, I had --experimental on, but tried it off (and in safe) as well.

I would always get past the new loading part at the bottom, although it would take a bit, but almost universally the process for a failure was this:

- Cold boot into windows.
- Launch nothing other than things that load automatically
- Run SD.next
- Launch edge and open the web page, wait until the info at the bottom had settled down
- Type any old text and load, as best as I could tell, any model and generate.
- Sometimes it would bomb instantly, sometimes at 20%, but always before finishing.

As mentioned I'm having no problems with tiger's A1111, so it must be something either imported from A1111 base than tiger has not brought in yet, or something specific to SD.next

**Comment by TeutonJon78** [comment]
In settings, what is are settings under Optimizations and Computer settings.

You generally shouldn't be running experimental unless you have a specific need to.

**Comment by Lumpsack** [comment]
Well, in safe experimental was removed so not really the problem, but I've taken it from my normal launch now.

Here are the settings 

![image](https://github.com/vladmandic/automatic/assets/127429289/e19ae42a-3d80-4056-b24b-0ed326d3ca89)
![image](https://github.com/vladmandic/automatic/assets/127429289/a6d7c20e-d517-4b82-b4b8-e29f9f723045)

After confirming it didn't work I upgraded and other than a new fancy logo on start no good news, it still bombs immediately.

**Comment by TeutonJon78** [comment]
Set the precision type to Full, Device to FP32, enable upcast cross attention layer, and TF32 math ops.

DirectML doesn't support autocast, so you need to force FP32 mode.  lshqqytigger added some support in (that needs the experimental flag), but it can be flaky and unreliable. The only time I've run into this error was when I was playing around in FP16 mode.

**Comment by Lumpsack** [comment]
Thanks for the advice, I tried those and it was briefly looking hopeful, I reached the lofty heights of 60%
So I tried with experimental on again, and back to insta stop.
Whatever change was made in the couple of days before I posted this really doesn't agree with my system, its a shame I wasn't generating daily so I could pin it down more.

**Comment by TeutonJon78** [comment]
You can make a secondary install to bisect the changes if you want to track it down.

After the `git clone ...` command issue a `git checkout SHA1_OF_THE_COMMIT_YOU_WANT` (which can get from the commit history on github) and it will force the code to the state at that moment. As long as you don't edit any files, you can trivially change around the commits. 

If you want to return it to the front of the repo, just issue `git checkout master` or `git checkout -f master` if you changed anything. Then you can just `git pull` like normal to keep it updated.

**Comment by vladmandic** [comment]
any updates here?

**Comment by Lumpsack** [comment]
Thanks for the follow up, no, still the same issue - I ran the upgrade three times to let it update, uninstall something not needed and settle, then ran in without upgrade - it got maybe 30% then failed with the same error.
Tigers still works fine under Windows and I've since installed A1111 on Linux (and God do I hate Linux - worst OS eva ;) )

**Comment by vladmandic** [comment]
cc @lshqqytiger since its pretty much his code here - and if his fork works, no reason why here it shouldn't.

**Comment by vladmandic** [comment]
this issue has been idle for a long time - and no other amd user has been to reproduce it. closing as idle, but can be reopened if further information is provided.

**Comment by Omniphantasm** [comment]
Not sure if this helps but I had same issue with DirectML, turned out that when I first installed it it installed regular Torch and not DirectML.Tried uninstalling torch and forcing install of DirectML and then got this error. Couldn't figure out how to fix it so I deleted entire folder, re-cloned the git and then before the very first run I used the --use-directml command line argument and problem didn't re-occur.

Don't know much but maybe something got corrupted that just wasn't fixed by simply uninstalling torch and installing the torch directml version.
